### PR TITLE
Edit form

### PR DESCRIPTION
**Pull request by raphaeldieu**
Features added:
- User can create a form on the dashboard
- User can edit a form : 
                                    - add content
                                    - display an edit panel for each content
- Add Service for user
- Deleted Service for user (???!!!)

lots of love

### CODE REVIEW

**Comment by ggharry** [comment]
Did you rebase yet?

**Comment by h8rry** [comment]
Did you see my comments in #16 ?

**Comment by ctlong** [comment]
Yes, we removed the status stuff because it meant that the response was always an error message. We have also removed the factory from the navbar file and the .pry from the contents controller.

File: app/assets/javascripts/angular/controller/forms/updateForms.js.erb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    };
+
+    $http.post(url + '/forms/' + $routeParams.id + '/contents',$scope.content).success(function(response) {
+    $scope.fields.push(response);
+    }).error(function(response) {
+    });
+  };
+
+  //Delete a content
+  $scope.delete = function(field){
```
Comment (h8rry) [inline app/assets/javascripts/angular/controller/forms/updateForms.js.erb:31]: You can delete based on the current index

File: app/assets/javascripts/angular/controller/forms/updateForms.js.erb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    $http.delete(url + '/contents/' + field.id).success(function(response){
+      $scope.fields.splice($scope.fields.indexOf(field),1)
```
Comment (h8rry) [inline app/assets/javascripts/angular/controller/forms/updateForms.js.erb]: Then you don't need to use `indexOf`

File: app/assets/javascripts/angular/controller/static_pages/navbar.js.erb [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
-
-  $auth.validateUser().then(function(resp) {
-    $scope.loggedIn  = true;
-    $scope.home = "dashboard"
-  }).catch(function(resp) {
+  
+  $auth.validateUser().then(function(response) {
     $scope.loggedIn  = false;
+  }).catch(function(response) {
+    $scope.loggedIn = true;
```
Comment (h8rry) [inline app/assets/javascripts/angular/controller/static_pages/navbar.js.erb:5]: If there's an error, then loggedIn is true?

File: app/controllers/submissions_controller.rb [hunk 4]
```diff
@@ -17,7 +17,7 @@ def destroy
       render json: {message: "Cannot find submission"}, status: :not_found
     else 
       if @submission.destroy
-        render json: {deleted: true}, status: :success
+        render json: {deleted: true}
```
Comment (h8rry) [inline app/controllers/submissions_controller.rb:20]: I already mentioned in #16 that you should put status in responses

Comment (ctlong) [inline app/controllers/submissions_controller.rb:20]: If we put the status in then the response is an error message, so we took it out for the success responses.

File: app/assets/javascripts/angular/controller/forms/updateForms.js.erb (hunk 1 above)
Comment (raphaeldieu) [inline app/assets/javascripts/angular/controller/forms/updateForms.js.erb:31]: fixed !

**Comment by ctlong** [comment]
Reviewed and edited.

File: app/controllers/submissions_controller.rb (hunk 4 above)
Comment (f3r) [inline app/controllers/submissions_controller.rb:20]: checkout [2.2.13.4 The :status Option](http://guides.rubyonrails.org/layouts_and_rendering.html#using-render) to know which status codes are available. And remember, always...

```
CONVENTION OVER CONFIGURATION
```

**Pull request merged**